### ISSUE TITLE
Feature request: resumable jobs

### ISSUE THREAD

**Issue by bbuchalter** [mufeedvh/pdfrip#17]
As a user,
I'd like to be able to stop a cracking job and restart it later,
So I can not lose progress.

**Comment by mufeedvh** [comment]
Hey @bbuchalter, thank you for the feature suggestion. This is a much-needed feature, will implement this in the next release! :raised_hands:

**Comment by Pommaq** [comment]
This can probably be done by expanding the current Producer interface to have "serialize" and "deserialize" functions (But I think it would be better to add a 2nd trait to avoid bloating). Then catching upon ctrl_c/other signals  we serialize the producer to disk. Finally just add a paremeter to resume operations or something.

I imagine the abort logic itself can be implemented using tools from Tokio.
Select! macro, channels, a small wrapper and a CancellationToken should be enough for this.

**Comment by mufeedvh** [comment]
@Pommaq, yes serializing the state to disk seems like the optimal approach here. Leaving some relevant links for future reference during implementation:
- https://criu.org/Main_Page
- https://github.com/checkpoint-restore/rust-criu
- https://github.com/bytecodealliance/wizer
- https://github.com/koute/speedy
